src/fs/driver/dvfs_driver.c

```diff
@@ -22,20 +22,20 @@ struct dvfs_driver_head {
 	struct dumb_fs_driver *drv;
 };
 
-ARRAY_SPREAD_DEF(struct dumb_fs_driver *, dumb_drv_tab);
+ARRAY_SPREAD_DEF(const struct dumb_fs_driver *const, dumb_drv_tab);
 
 struct dumb_fs_driver *dumb_fs_driver_find(const char *name) {
-	struct dumb_fs_driver *fs_drv;
+	const struct dumb_fs_driver *fs_drv;
 	array_spread_foreach(fs_drv, dumb_drv_tab) {
 		if (!strcmp(name, fs_drv->name)) {
-			return fs_drv;
+			return (struct dumb_fs_driver *)fs_drv;
```

Comment:
This doesn't look like a good idea. Either the structure is indeed immutable, as declared, and then the lookup function must propagate the `const` modifier, or otherwise the array elements must not be declared `const`.